src/main/java/org/kuali/kfs/sys/document/service/impl/AccountingLineRuleHelperServiceImpl.java

```diff
@@ -0,0 +1,589 @@
+            String errorPropertyIdentifyingName, boolean useShortMessages) {
+        String label = getSubObjectCodeLabel();
+
+        // Check exists & active
+        // CORNELL FIX: boolean isActive = subObjectCode != null && subObjectCode.isActive();
+        boolean isActive = ObjectUtils.isNotNull(subObjectCode) && subObjectCode.isActive();
+        return getExistsActiveValidationErrors(subObjectCode, isActive, useShortMessages, errorPropertyName, errorPropertyIdentifyingName, label);
+    }
+
+    /**
+     * Validates an accounting line. Side effects: refreshes referenced BOs and sets global error messages
+     *
+     * @param accountingLine
+     * @return boolean whether the accounting line is valid
+     */
+    @Override
+    public boolean validateAccountingLine(AccountingLine accountingLine) {
+        if (accountingLine == null) {
+            throw new IllegalStateException(configurationService.getPropertyValueAsString(
+                    KFSKeyConstants.ERROR_DOCUMENT_NULL_ACCOUNTING_LINE));
+        }
+
+        // Refresh reference objects
+        accountingLine.refreshReferenceObject("chart");
+        accountingLine.refreshReferenceObject("account");
+        accountingLine.refreshReferenceObject("objectCode");
+        if (StringUtils.isNotBlank(accountingLine.getSubAccountNumber())
+                && !accountingLine.getSubAccountNumber().equals(getDashSubAccountNumber())) {
+            accountingLine.refreshReferenceObject("subAccount");
+        }
+        if (StringUtils.isNotBlank(accountingLine.getFinancialSubObjectCode())) {
+            accountingLine.refreshReferenceObject("subObjectCode");
+        }
+        if (StringUtils.isNotBlank(accountingLine.getProjectCode())) {
+            accountingLine.refreshReferenceObject("project");
+        }
+        if (StringUtils.isNotBlank(accountingLine.getReferenceOriginCode())) {
+            accountingLine.refreshReferenceObject("referenceOrigin");
+        }
+
+        List<AccountingLineValidationError> validationErrors = getAccountingLineValidationErrors(accountingLine, false);
+        setGlobalErrorMessages(validationErrors);
+
+        boolean valid = validationErrors.isEmpty();
+
+        valid &= hasRequiredOverrides(accountingLine, accountingLine.getOverrideCode());
+        return valid;
+    }
+
+    @Override
+    public List<AccountingLineValidationError> getAccountingLineValidationErrors(AccountingLine accountingLine, boolean useShortMessages) {
+        List<AccountingLineValidationError> validationErrors = new ArrayList<>();
+
+        //get the accounting line sequence string to identify which line has error.
+        String accountIdentifyingPropertyName = getAccountIdentifyingPropertyName(accountingLine);
+
+        // retrieve accounting line objects to validate
+        Chart chart = accountingLine.getChart();
+        Account account = accountingLine.getAccount();
+        ObjectCode objectCode = accountingLine.getObjectCode();
```

Comment:
Nancy mentioned the possibility of adding some non-null checks for this section of the code. However, since this particular method relies heavily on validation code that requires a non-null accounting line, it might be better to just have this method return an empty validation errors list if the accounting line is null.

Reply:
Adjusting as recommended.